### ISSUE TITLE
MySQL error in UpdateClient method

### ISSUE THREAD

**Issue by wuleninja** [felipeweb/osin-mysql#7]
Hi,

First: thanks for your work !

I think I've found an error the 'UpdateClient' method:

```
	if _, err := s.db.Exec(fmt.Sprintf("UPDATE %sclient SET (secret, redirect_uri, extra) = (?, ?, ?) WHERE id=?", s.tablePrefix), c.GetSecret(), c.GetRedirectUri(), data, c.GetId()); err != nil {
```

Shouldn't the UPDATE statement be like: "set secret = ?, redirect_uri = ?, extra = ?" ?

Thanks again.

**Comment by felipeweb** [comment]
Thank you do you want to send a pull request?

On Sun, 29 Jan 2017 at 20:26 ninjawule <notifications@github.com> wrote:

Hi,

First: thanks for you work !

I think I've found an error the 'UpdateClient' method:

	if _, err := s.db.Exec(fmt.Sprintf("UPDATE %sclient SET (secret,
redirect_uri, extra) = (?, ?, ?) WHERE id=?", s.tablePrefix),
c.GetSecret(), c.GetRedirectUri(), data, c.GetId()); err != nil {

Shouldn't the UPDATE statement be like: "set secret = ?, redirect_uri = ?,
extra = ?" ?

Thanks again.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/felipeweb/osin-mysql/issues/7>, or mute the thread

**Comment by wuleninja** [comment]
Hi ! Thanks for your answer !

I think I've solved it. But I cannot push a correction branch yet. 

Anyway, the correction is pretty straightforward:
 in mysql.go:103, replacing:

```
SET (secret, redirect_uri, extra) = (?, ?, ?) WHERE
```

with 

```
SET secret=?, redirect_uri=?, extra=? WHERE
```

**Comment by felipeweb** [comment]
You can send a WIP pull request and use Travis to see witch test are
failing

**Comment by wuleninja** [comment]
I'm quite a noob at using GitHub. Do I have to fork your project beforehand ?
I know why the tests are failing on my machine. It's just because my 'root' user is password protected.

**Comment by felipeweb** [comment]
don't worry maybe this link
https://guides.github.com/activities/contributing-to-open-source/ help you

Em seg, 30 de jan de 2017 às 11:40, ninjawule <notifications@github.com>
escreveu:

**Comment by felipeweb** [comment]
Closes with #8 thank you @ninjawule
